app/views/posts/index.html.erb

```diff
@@ -1 +1,38 @@
-<h1>All posts</h1>
\ No newline at end of file
+<div class="d-flex justify-content-between gap-2 row post-row-mt">
+  <div class="card width-25 col-4">
+    <%= image_tag(@user.photo, alt: 'User Photo') if @user.photo.present? %>
+  </div>
+
+  <div class="card flex-row justify-content-between align-items-baseline gap-2 col-8">
+    <h2><%= @user.name %></h2>
+    <p>Number of posts <%= @user.posts_counter %></p>
+  </div>
+</div>
+
+<% @posts.each do |post| %>
+  <div class="row gap-1 justify-content-center margin-tp-2">
+    <div class="card">
+      <div class="row">
+        <h2><%= link_to post.title, user_post_path(@user, post) %></h2>
+        <p><%= post.text %></p>
+      </div>
+      <div class="row">
+        <p class="d-flex justify-content-end align-items-end">comments: <%= post.comments.count %> likes: 3</p>
+      </div>
+    </div>
+    <div class="card">
+      <h2>Comments</h2>
+      <% post.comments.each do |comment| %>
+        <div class="comment">
+          <p><strong><%= comment.user.name %>:</strong> <%= comment.text %></p>
+        </div>
+      <% end %>
```

Comment:
👏 Excellent progress on your project! To further enhance the functionality and optimize resource usage, consider the following improvements:

- [ ] **Displaying Recent Comments:** Currently, your controller retrieves all comments for a post, and in the HTML file, you are displaying all of them. However, as per the wireframe, you should only show the most recent 5 comments. I recommend updating your code to ensure it functions as specified in the wireframe.
![image](https://github.com/NoelLincoln/blog-app/assets/55840999/a56e7cb8-2246-42b9-ab6b-95f78dcd16c3)

- [ ] [Optioanl] **Optimizing Comment Retrieval:** Additionally, your current approach involves retrieving all comments for a post, which can be resource-intensive, especially for posts with a large number of comments. Instead, please utilize the `recent_comments` method in your `Post` model. This method efficiently fetches the latest 5 comments, optimizing resource usage by fetching only the necessary data.

By implementing these improvements, you'll ensure a more informative and efficient user experience. Keep up the great work! 🚀👏